Unable to build apm under Node v7.0.0

**Description**:
When attempting to build Atom from source, the process fails out when installing apm. Probably related to #635 since I can build just fine if I revert to node v6.9.1.

**Steps to Reproduce:**
1. Have node v7.0.0 installed
2. Fresh clone of atom
3. Run `script/build`

**Stacktrace**:

```
child_process.js:145
    throw new TypeError('Incorrect value of args option');
    ^

TypeError: Incorrect value of args option
    at Object.exports.execFile (child_process.js:145:11)
    at Object.exports.exec (child_process.js:108:18)
    at Object.<anonymous> (/home/dnoel/src/atom/apm/node_modules/atom-package-manager/script/postinstall.js:10:16)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.runMain (module.js:607:10)
    at run (bootstrap_node.js:382:7)
npm ERR! Linux 4.8.6-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "--global-style" "--loglevel=error" "install"
npm ERR! node v7.0.0
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
    at checkExecSyncError (child_process.js:488:13)
    at Object.execFileSync (child_process.js:508:13)
    at module.exports (/home/dnoel/src/atom/script/lib/install-apm.js:10:16)
    at Object.<anonymous> (/home/dnoel/src/atom/script/bootstrap:27:1)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.require (module.js:500:17)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,6 @@ notifications:
     on_failure: change
 
 node_js:
-  - "0.10"
   - "4"
   - "6"
   - "7"

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ node_js:
   - "0.10"
   - "4"
   - "6"
+  - "7"
 
 script: ./bin/npm test
 

--- a/script/postinstall.js
+++ b/script/postinstall.js
@@ -7,6 +7,6 @@ if (process.platform.indexOf('win') === 0) {
 } else {
   script += '.sh'
 }
-var child = cp.spawn(script, [], {stdio: ['pipe', 'pipe', 'pipe']})
+var child = cp.spawn(script, [], { stdio: ['pipe', 'pipe', 'pipe'], shell: true })
 child.stderr.pipe(process.stderr)
 child.stdout.pipe(process.stdout)

--- a/script/postinstall.js
+++ b/script/postinstall.js
@@ -7,6 +7,6 @@ if (process.platform.indexOf('win') === 0) {
 } else {
   script += '.sh'
 }
-var child = cp.exec(script, [], {stdio: ['pipe', 'pipe', 'pipe']})
+var child = cp.spawn(script, [], {stdio: ['pipe', 'pipe', 'pipe']})
 child.stderr.pipe(process.stderr)
 child.stdout.pipe(process.stdout)

